lib/std/mem.zig

```diff
@@ -1668,12 +1668,22 @@ pub const SplitIterator = struct {
 /// Naively combines a series of slices with a separator.
 /// Allocates memory for the result, which must be freed by the caller.
 pub fn join(allocator: *Allocator, separator: []const u8, slices: []const []const u8) ![]u8 {
+    return joinMaybeZ(allocator, separator, slices, false);
+}
+
+/// Naively combines a series of slices with a separator and null terminator.
+/// Allocates memory for the result, which must be freed by the caller.
+pub fn joinZ(allocator: *Allocator, separator: []const u8, slices: []const []const u8) ![]u8 {
```

Comment:
```suggestion
pub fn joinZ(allocator: *Allocator, separator: []const u8, slices: []const []const u8) ![:0]u8 {
```

It might be cleaner to keep the implementations separate given that the null termination is part of the type

Reply:
Well that was rather dumb, thanks for the catch! I'll see how the implementation works out, you may be right.